app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -121,7 +121,7 @@ echo "#"
 
 <% if plugin_present?('katello') && activation_keys.present? -%>
 register_katello_host(){
-    UUID=$(subscription-manager identity | head -1 | awk '{print $3}')
+    UUID=$(subscription-manager identity | sed -ne 's,.*\([[:xdigit:]]\{8\}-[[:xdigit:]]\{4\}-[[:xdigit:]]\{4\}-[[:xdigit:]]\{4\}-[[:xdigit:]]\{12\}\).*,\1,p')
```

Comment:
gnu grep should have `-m NUM` which could work here. It counts matching lines instead of matches, but hopefully we can rely on subman not printing more than one uuid per line.